### PR TITLE
test: 78% coverage of split-expense

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=0PjYGEm)
### <samp>🤖 Generated by Copilot at 7a9888f</samp>

This pull request adds test cases and mock data for the split expense page component in the app. The test cases verify the logic of the `onChangeAmount` and `onChangePercentage` methods using the mock data. The mock data simulates different split expense scenarios with various form values.

### <samp>🤖 Generated by Copilot at 7a9888f</samp>

> _We mock the data for the split expense_
> _We test the methods with different inputs_
> _We face the doom of the currency change_
> _We resist the errors with our outputs_

### <samp>🤖 Generated by Copilot at 7a9888f</samp>

*  Add mock data for split expense form groups ([link](https://github.com/fylein/fyle-mobile-app/pull/2235/files?diff=unified&w=0#diff-34a64d3b11f026ed5059cb78801436ea312cfab12781dbd085479cae1817ad48R1-R49))
*  Import mock data for testing split expense page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2235/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5R120-R127))
*  Test `onChangeAmount` and `onChangePercentage` methods for different scenarios and expected outcomes ([link](https://github.com/fylein/fyle-mobile-app/pull/2235/files?diff=unified&w=0#diff-391b3c1134d2ffdd6f0128e3e95116db167b57a91a80674dbb0abfe568407ec5L342-R439))

## Clickup
https://app.clickup.com/t/85ztjmt7v

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 1]
```diff
@@ -339,6 +347,94 @@ describe('SplitExpensePage', () => {
       expect(otherSplitExpenseForm.controls.percentage.value).toEqual(4);
       expect(component.getTotalSplitAmount).toHaveBeenCalledTimes(1);
     }));
+
+    it('should return void immediately if amount is null', () => {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:351]: add the related tests inside a separate `describe` block

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 2]
```diff
@@ -339,6 +347,94 @@ describe('SplitExpensePage', () => {
...
+    it('should return void immediately if splitExpenseForm.amount is not a number', () => {
+      spyOn(component, 'getTotalSplitAmount');
+      component.amount = 2000;
+      const splitExpenseForm1 = cloneDeep(splitExpenseFormData2);
+
+      Object.defineProperty(splitExpenseForm1.controls.amount, '_pendingChange', { value: true });
+
+      component.onChangeAmount(splitExpenseForm1, 0);
+
+      expect(splitExpenseForm1.controls.amount.value).toEqual(null);
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:373]: `toBeNull()`

**Review by Dimple16 — APPROVED** [review]
minor comments

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:351]: These tests are part of `onChangeAmount` method therefore keeping in the respective describe block.

**Pull request merged**